source/scripts/model/Monster.js

```diff
@@ -32,15 +36,14 @@ export default class Monster {
         this.animation = false
 
         if(this.phase == true) {
-            var dx = this.game.adventurer.position.x - this.position.x
-            var dy = this.game.adventurer.position.y - this.position.y
-
-            if(Math.abs(dx) > Math.abs(dy)) {
-                if(dx > 0) this.move({x: +1})
-                if(dx < 0) this.move({x: -1})
-            } else {
-                if(dy > 0) this.move({y: +1})
-                if(dy < 0) this.move({y: -1})
+            switch(this.movement) {
+            case "simplechase":
+                this.move(Movement.getSimpleChaseMovement(this.game.adventurer.position, this.position))
+                break
+            case "wander":
+            case "wander-orthogonal":
+            case "wander-diagonal":
+                this.move(Movement.getWanderMovement(Movement[this.movement.slice(7)].apply()))
```

Comment:
If we can split this logic up into separate lines, that'd make it much more readable.